sdk/eventgrid/Azure.ResourceManager.EventGrid/src/Generated/SubscriptionResource.cs

```diff
@@ -0,0 +1,298 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.ResourceManager;
+using Azure.ResourceManager.EventGrid.Models;
+
+namespace Azure.ResourceManager.EventGrid
+{
+    /// <summary>
+    /// A Class representing a Subscription along with the instance operations that can be performed on it.
+    /// If you have a <see cref="ResourceIdentifier" /> you can construct a <see cref="SubscriptionResource" />
+    /// from an instance of <see cref="ArmClient" /> using the GetSubscriptionResource method.
+    /// Otherwise you can get one from its parent resource <see cref="NamespaceTopicResource" /> using the GetSubscription method.
+    /// </summary>
+    public partial class SubscriptionResource : ArmResource
```

Comment:
We need to rename `SubscriptionResource` to a better name, it is conflicting with [Azure.ResourceManager.Resources.SubscriptionResource](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/resourcemanager/Azure.ResourceManager/src/Resources/Generated/SubscriptionResource.cs), and also [EventGridSubscriptionData](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventgrid/Azure.ResourceManager.EventGrid/src/Generated/EventGridSubscriptionData.cs) is in use.

Add below config under rename-mapping in autorest.md
```
Subscription: ABetterNameBasedOnTheUsage
```

Reply:
fixed

Reply:
This file still got generated, not resolved.
And it is causing the conflicts with [Azure.ResourceManager.Resources.SubscriptionResource](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/resourcemanager/Azure.ResourceManager/src/Resources/Generated/SubscriptionResource.cs), also generating wrong operation methods in `EventGridExtensions.cs`, ending up to build errors.

Reply:
fixed